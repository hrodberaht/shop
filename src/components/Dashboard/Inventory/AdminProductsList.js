import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ConnectedAdminProduct from './AdminProduct';
import { getProductsAll, getProductsLoaded } from '../../../store/products/selectors';
import { fetchProducts } from '../../../store/products/actionCreators';
import { getAuthToken } from '../../../store/authenticate/selectors';

export class AdminProductsList extends Component {
  componentDidMount() {
    const { getProducts, token } = this.props;
    getProducts(token);
  }

  render() {
    const { products, loaded } = this.props;
    if (!loaded) return <p>Loading...</p>;
    return (
      <div className="admin-product__list">
        <table>
          <tbody>
            <tr>
              <th>Name:</th>
              <th>Image:</th>
              <th>Type:</th>
              <th>Price:</th>
              <th>Stock:</th>
              <th>Edit:</th>
              <th>Remove:</th>
            </tr>
            {products.map(product => (
              <tr key={product.id}>
                <ConnectedAdminProduct product={product} />
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  products: getProductsAll(state),
  token: getAuthToken(state),
  loaded: getProductsLoaded(state),
});

export default connect(
  mapStateToProps,
  {
    getProducts: fetchProducts,
  },
)(AdminProductsList);

AdminProductsList.propTypes = {
  products: PropTypes.arrayOf(PropTypes.object).isRequired,
  getProducts: PropTypes.func.isRequired,
  token: PropTypes.string.isRequired,
  loaded: PropTypes.bool.isRequired,
};
