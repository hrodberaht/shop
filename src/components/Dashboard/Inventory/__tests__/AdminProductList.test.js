import { AdminProductsList } from '../AdminProductsList';

const defaultProps = {
  products: [
    {
      id: 'product-1',
      type: 'Printer',
      name: 'ZX3',
      price: '400',
      inStock: '30',
    },
    {
      id: 'product-2',
      type: 'Printer',
      name: 'ZX8',
      price: '800',
      inStock: '40',
    },
  ],
  token: '1234',
  loaded: false,
  getProducts: jest.fn(),
};

const setup = buildSetup(AdminProductsList, defaultProps);

describe('<AdminProductsList />', () => {
  it('should render loading component if loaded false', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render list product component if loaded true', () => {
    const { wrapper } = setup();
    wrapper.setProps({ loaded: true });
    expect(wrapper).toMatchSnapshot();
  });
  it('should call getProducts if componentDiDMount', () => {
    const { wrapper } = setup();
    expect(wrapper.instance().props.getProducts).toHaveBeenCalled();
  });
});
