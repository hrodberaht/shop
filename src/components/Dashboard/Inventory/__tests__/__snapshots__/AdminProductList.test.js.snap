// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminProductsList /> should render list product component if loaded true 1`] = `
<div
  className="admin-product__list"
>
  <table>
    <tbody>
      <tr>
        <th>
          Name:
        </th>
        <th>
          Image:
        </th>
        <th>
          Type:
        </th>
        <th>
          Price:
        </th>
        <th>
          Stock:
        </th>
        <th>
          Edit:
        </th>
        <th>
          Remove:
        </th>
      </tr>
      <tr
        key="product-1"
      >
        <Connect(AdminProduct)
          product={
            Object {
              "id": "product-1",
              "inStock": "30",
              "name": "ZX3",
              "price": "400",
              "type": "Printer",
            }
          }
        />
      </tr>
      <tr
        key="product-2"
      >
        <Connect(AdminProduct)
          product={
            Object {
              "id": "product-2",
              "inStock": "40",
              "name": "ZX8",
              "price": "800",
              "type": "Printer",
            }
          }
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<AdminProductsList /> should render loading component if loaded false 1`] = `
<p>
  Loading...
</p>
`;
